packages/server-wallet/server-wallet.api.md

```diff
@@ -235,6 +253,28 @@ export type NetworkConfiguration = {
     chainNetworkID: number;
 };
 
+// @public (undocumented)
+export type ObjectiveDoneResult = ObjectiveSuccess | ObjectiveError;
+
+// @public (undocumented)
+export type ObjectiveError = EnsureObjectiveFailed | InternalError;
+
+// @public
+export type ObjectiveResult = {
+    done: Promise<ObjectiveDoneResult>;
+    currentStatus: ObjectiveStatus;
+    objectiveId: string;
+    channelId: string;
```

Comment:
Have you considered providing the `channelId` in the `ObjectiveDoneResult`?

Currently, a developer could
```
const {done, channelId} = await wallet.createChannels(...);
await done;
await wallet.updateChannel({channelId, ...});
```

resulting in a runtime error in the case that `done` rejects.

We could potentially nudge them into more robust error handling, at least in Typescript, with a flow like

```
const {done, objectiveId} = await wallet.createChannels(...);
const result = await done;
switch (result.type) {
  case 'ObjectiveSuccess':
    await wallet.updateChannel({channelId: result.channelId, ...});
    break;
  case 'ObjectiveFailure':
    await handleError(result.error);
    break;
}
```

Reply:
That's a good point!

I did consider this a little bit but I wasn't sure if it would be strange to include the `channelId` on the `objective` as well as the `ObjectiveDoneResult`.

However, it does make the flow a lot nicer so I think it makes sense!

Reply:
Just to be clear, I'm suggesting _not_ including the channelId in the result of `createChannels`. This forces them to `await done` in order to access the channelId.

(I think it's valid to leave the responsibility of checking the type of the return promise up to the developer -- the above question is more open-ended.)